rfcs/text/0012_background_sessions.md

```diff
@@ -0,0 +1,291 @@
+queries, etc.). When running slower searches on "cold" storage or frozen indices, users encounter timeouts after 30
+seconds by default.
+
+In 7.7, with the introduction of the `_async_search` API in Elasticsearch, we provided Kibana users a way to bypass the
+timeout, but users still need to remain on-screen for the entire duration of the search requests.
+
+By default, when a user navigates away before search requests finish, we cancel them. This is a good default behavior,
+since there currently isn't any way for the user to later return to the results.
+
+The primary motivation of this RFC is to enable users to do the following without needing to keep Kibana open, or while
+moving onto other work inside Kibana:
+
+- Run long search requests (beyond 30 seconds)
+- View their status (complete/incomplete)
+- Cancel incomplete search requests
+- Retrieve completed search request results 
+
+# Detailed design
+
+Because a single view (such as a dashboard with multiple visualizations) can initiate multiple search requests, we need
+a way to associate the search requests together in a single entity. We call this entity a `session`, and when a user
+decides that they want to continue running the search requests while moving onto other work, we will create a saved
+object corresponding with that specific `session`.
+
+We will expose a new `session` service inside of the `data` plugin `search` service. The service will expose APIs for 
+the following:
+
+- Starting a new session (generating a unique identifier for this specific grouping of search requests)
+- Saving a session (creating the saved object using that ID)
+- Getting a list of sessions for the current user (including status, any necessary application data, and a URL to
+restore the view)
+- Cancelling an incomplete session (deleting the saved object and cancelling any incomplete search requests)
+- Restoring a session (allowing search requests to resume instead of restart)
+
+We will also modify the existing `search` API exposed by the `data` plugin `search` service to optionally accept an
+additional parameter, the `sessionId`, which will allow search requests to *resume* instead of *restart*. When
+"restoring" a session, it is crucial that search requests are sent with *identical* search request parameters as were
+sent originally. (For example, relative date ranges should be converted to absolute before passing the request to the
+`search` service.)
+
+Applications query data from Elasticsearch on behalf of the end user in a variety of ways:
+
+- Using `SearchSource` (our higher-level abstracted search API, used by Maps, Lens, and most visualizations, exposed by
+the `data` plugin `search` service)
+- Using `search` (our lower-level search API that accepts raw Elasticsearch query DSL, used by Vega and Timelion,
+exposed by the `data` plugin `search` service)
+- Exposing a route that bypasses our `search` services and queries Elasticsearch directly, then manipulates the data
```

Comment:
should we worry about this one ? if you are bypassing our search services you should first start using our search service. So i think we only want to address the case when you are exposing a route that internally uses our server side search.